server/channels/app/syncables.go

```diff
@@ -154,66 +158,75 @@ func (a *App) DeleteGroupConstrainedMemberships(c request.CTX) error {
 // deleteGroupConstrainedTeamMemberships deletes team memberships of users who aren't members of the allowed
 // groups of the given group-constrained team. If a teamID is given then the procedure is scoped to the given team,
 // if teamID is nil then the procedure affects all teams.
-func (a *App) deleteGroupConstrainedTeamMemberships(c request.CTX, teamID *string) error {
+func (a *App) deleteGroupConstrainedTeamMemberships(rctx request.CTX, teamID *string) error {
 	teamMembers, appErr := a.TeamMembersToRemove(teamID)
 	if appErr != nil {
 		return appErr
 	}
 
+	var multiErr *multierror.Error
 	for _, userTeam := range teamMembers {
-		err := a.RemoveUserFromTeam(c, userTeam.TeamId, userTeam.UserId, "")
-		if err != nil {
-			return err
-		}
-
-		c.Logger().Info("removed teammember",
+		logger := rctx.Logger().With(
 			mlog.String("user_id", userTeam.UserId),
 			mlog.String("team_id", userTeam.TeamId),
 		)
+
+		err := a.RemoveUserFromTeam(rctx, userTeam.TeamId, userTeam.UserId, "")
+		if err != nil {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("failed to add team member for default team membership: %w", err))
+			continue
+		}
+
+		logger.Info("Removed team member for group contrained team membership")
 	}
 
-	return nil
+	return multiErr.ErrorOrNil()
 }
 
 // deleteGroupConstrainedChannelMemberships deletes channel memberships of users who aren't members of the allowed
 // groups of the given group-constrained channel. If a channelID is given then the procedure is scoped to the given team,
 // if channelID is nil then the procedure affects all teams.
-func (a *App) deleteGroupConstrainedChannelMemberships(c request.CTX, channelID *string) error {
+func (a *App) deleteGroupConstrainedChannelMemberships(rctx request.CTX, channelID *string) error {
 	channelMembers, appErr := a.ChannelMembersToRemove(channelID)
 	if appErr != nil {
 		return appErr
 	}
 
+	var multiErr *multierror.Error
 	for _, userChannel := range channelMembers {
-		channel, err := a.GetChannel(c, userChannel.ChannelId)
+		logger := rctx.Logger().With(
+			mlog.String("user_id", userChannel.UserId),
+			mlog.String("channel_id", userChannel.ChannelId),
+		)
+
+		channel, err := a.GetChannel(rctx, userChannel.ChannelId)
 		if err != nil {
-			return err
+			multiErr = multierror.Append(multiErr, fmt.Errorf("failed to get channel for group contrained channel membership: %w", err))
```

Comment:
typo:
```suggestion
			multiErr = multierror.Append(multiErr, fmt.Errorf("failed to get channel for group constrained channel membership: %w", err))
```

Replace other `contrained` as well in the file.